xonsh/execer.py

```diff
@@ -178,7 +178,9 @@ def _parse_ctx_free(self, input, mode='exec'):
                     last_error_line = last_error_col = -1
                     input = '\n'.join(lines)
                     continue
-                if last_error_line > 1 and lines[idx-1].rstrip()[-1] == ':':
+
+                stripped_line = lines[idx-1].rstrip()
+                if last_error_line > 1 and stripped_line and stripped_line[-1] == ':':
```

Comment:
Isn't that the exact same line as before:

    if last_error_line > 1 and lines[idx-1].rstrip()[-1] == ':':

am I missing something here? :)